### PR TITLE
feat: support Mock SMS provider

### PR DESCRIPTION
**Pull request by kofj**
1. Update go-sms-sender to v0.3.0.
2. Fix: avoid page crash if not found provider info.

Signed-off-by: 疯魔慕薇 <kfanjian@gmail.com>

Fix: https://github.com/casdoor/casdoor/issues/1002

### CODE REVIEW

**Comment by kofj** [comment]
#1002

**Comment by kofj** [comment]
@hsluoyz Review pls, thx.

File: web/src/auth/Provider.js [hunk 1]
```diff
@@ -137,7 +137,8 @@ export function getProviderUrl(provider) {
 
     return `${urlObj.protocol}//${host}`;
   } else {
-    return Setting.OtherProviderInfo[provider.category][provider.type].url;
+    const info = Setting.OtherProviderInfo[provider.category][provider.type];
+    if (info) {return info.url;}
```
Comment (hsluoyz) [inline web/src/auth/Provider.js:141]: @qianxi0410 why the linter doesn't report this?

File: web/src/Setting.js [hunk 2]
```diff
@@ -532,7 +536,8 @@ export function getProviderLogoURL(provider) {
     }
     return `${StaticBaseUrl}/img/social_${provider.type.toLowerCase()}.png`;
   } else {
-    return OtherProviderInfo[provider.category][provider.type].logo;
+    const info = OtherProviderInfo[provider.category][provider.type];
+    if (info) {return info.logo;}
```
Comment (hsluoyz) [inline web/src/Setting.js:540]: @qianxi0410 why the linter doesn't report this?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/Setting.js (hunk 2 above)
Comment (hsluoyz) [reply web/src/Setting.js:540]: @kofj you already added new item to `OtherProviderInfo`, why add `if` here?

File: web/src/auth/Provider.js (hunk 1 above)
Comment (hsluoyz) [reply web/src/auth/Provider.js:141]: @kofj you already added new item to `OtherProviderInfo`, why add `if` here?

File: web/src/Setting.js (hunk 2 above)
Comment (kofj) [reply web/src/Setting.js:540]: In other cases maybe cause page crash, so we need check it first.

File: web/src/auth/Provider.js (hunk 1 above)
Comment (kofj) [reply web/src/auth/Provider.js:141]: In other cases maybe cause page crash, so we need check it first.

Comment (hsluoyz) [reply web/src/auth/Provider.js:141]: Code format should be:

```js
if (info) {
  return info.url;
}
```

And make sure all paths return value

File: web/src/Setting.js (hunk 2 above)
Comment (hsluoyz) [reply web/src/Setting.js:540]: same as above

File: web/src/auth/Provider.js (hunk 1 above)
Comment (kofj) [reply web/src/auth/Provider.js]: resolved

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**